### PR TITLE
Support `sprockets` v4

### PR DESCRIPTION
**Pull request by orien**
Add support for `sprockets` 4.

### CODE REVIEW

File: config/initializers/assets.rb [hunk 1]
```diff
@@ -1 +1 @@
-Rails.application.config.assets.precompile += ['guide/application.js', 'guide/scenario.js']
+Rails.application.config.assets.precompile += %w[guide/application.js guide/scenario.js guide/application.css]
```
Comment (orien) [inline config/initializers/assets.rb:1]: The only change required is explicitly reference the CSS included with the gem.

File: Appraisals [hunk 2]
```diff
@@ -1,6 +1,6 @@
 appraise 'rails-6.1' do
   gem 'railties', '~> 6.1.0'
-  gem 'sprockets', '~> 3'
+  gem 'sprockets', '~> 4'
```
Comment (orien) [inline Appraisals:3]: Use Sprockets 4 with Rails 6.1 in the test suite.

File: spec/test_apps/shared/app/assets/config/manifest.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+//= link_tree ../images
+//= link_directory ../javascripts .js
+//= link_directory ../stylesheets .css
```
Comment (orien) [inline spec/test_apps/shared/app/assets/config/manifest.js:3]: The test app needs a manifest file to be compatible with Sprockets 4.

**Review by ananova — APPROVED** [review]

File: config/initializers/assets.rb (hunk 1 above)
Comment (petervandoros) [reply config/initializers/assets.rb:1]: Is this how gems are meant to configure assets they use with sprockets, or does sprockets support `app/assets/config/manifest.js` file for gems as well?

**Review by petervandoros — COMMENTED** [review]
Belated 👍  with a question.

File: config/initializers/assets.rb (hunk 1 above)
Comment (orien) [reply config/initializers/assets.rb:1]: Good question.

I see in the Rails Guide it configures things a little differently:

```ruby
initializer "blorgh.assets.precompile" do |app|
  app.config.assets.precompile += %w( admin.js admin.css )
end
```
https://guides.rubyonrails.org/engines.html#separate-assets-and-precompiling

I'll try moving to this form.

**Pull request merged**